bus-mapping/src/evm/opcodes/create.rs

```diff
@@ -1,35 +1,43 @@
-use crate::circuit_input_builder::{CircuitInputStateRef, ExecStep};
-use crate::evm::Opcode;
-use crate::operation::{AccountField, AccountOp, CallContextField, TxAccessListAccountOp, RW};
-use crate::Error;
-use eth_types::{evm_types::gas_utils::memory_expansion_gas_cost, GethExecStep, ToWord, Word};
-use keccak256::EMPTY_HASH;
+use crate::{
+    circuit_input_builder::{
+        CircuitInputStateRef, CopyDataType, CopyEvent, ExecStep, NumberOrHash,
+    },
+    evm::Opcode,
+    operation::{AccountField, AccountOp, CallContextField, MemoryOp, RW},
+    Error,
+};
+use eth_types::{
+    evm_types::gas_utils::memory_expansion_gas_cost, Bytecode, GethExecStep, ToBigEndian, ToWord,
+    Word, H160, H256,
+};
+use ethers_core::utils::{get_create2_address, keccak256, rlp};
 
 #[derive(Debug, Copy, Clone)]
-pub struct DummyCreate<const IS_CREATE2: bool>;
+pub struct Create<const IS_CREATE2: bool>;
 
-impl<const IS_CREATE2: bool> Opcode for DummyCreate<IS_CREATE2> {
+impl<const IS_CREATE2: bool> Opcode for Create<IS_CREATE2> {
     fn gen_associated_ops(
         state: &mut CircuitInputStateRef,
         geth_steps: &[GethExecStep],
     ) -> Result<Vec<ExecStep>, Error> {
-        // TODO: replace dummy create here
         let geth_step = &geth_steps[0];
 
         let offset = geth_step.stack.nth_last(1)?.as_usize();
         let length = geth_step.stack.nth_last(2)?.as_usize();
 
-        let curr_memory_word_size = (state.call_ctx()?.memory.len() as u64) / 32;
+        let curr_memory_word_size = state.call_ctx()?.memory_word_size();
         if length != 0 {
             state
                 .call_ctx_mut()?
                 .memory
                 .extend_at_least(offset + length);
         }
-        let next_memory_word_size = (state.call_ctx()?.memory.len() as u64) / 32;
+        let next_memory_word_size = state.call_ctx()?.memory_word_size();
 
         let mut exec_step = state.new_step(geth_step)?;
 
+        let callee = state.parse_call(geth_step)?;
```

Comment:
the above line "let mut exec_step = state.new_step(geth_step)?;" should be put before memory expansion. Or mem_size inside exec_step will be incorrect